Makefile

```diff
@@ -78,6 +78,9 @@ codegen: ## Generate code. Must be run if changes are made to ./pkg/apis/...
 		output:crd:artifacts:config=charts/karpenter/crds
 	hack/boilerplate.sh
 
+docgen: ## Generate docs
```

Comment:
can you update the .PHONY list. it looks like we're missing a few of the other ones like "nightly", "snapshot", and "stablerelease" as well.